pkg/fqdn/helpers.go

```diff
@@ -99,10 +103,15 @@ func injectToCIDRSetRules(rule *api.Rule, cache *DNSCache, reMap *regexpmap.Rege
 					}).Debug("Emitting matching DNS Name -> IPs for ToFQDNs Rule")
 					delete(missing, ToFQDN.MatchPattern)
 					emitted[name] = append(emitted[name], ips...)
-					egressRule.ToCIDRSet = append(egressRule.ToCIDRSet, api.IPsToCIDRRules(ips)...)
+					allIPs = append(allIPs, ips...)
 				}
 			}
 		}
+
+		// Only overwrite ToCIDRSet for rules with FQDN elements
+		if len(allIPs) > 0 {
+			egressRule.ToCIDRSet = api.IPsToCIDRRules(ip.KeepUniqueIPs(allIPs))
```

Comment:
oh, one last thing. The comment on`injectToCIDRSetRules` already existing rules are preserved. We should remove that line (44) above.